src/Perpetuum/Services/EventServices/EventProcessors/DayTimeEventProcessor.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using Perpetuum.ExportedTypes;
+using Perpetuum.Services.Daytime;
+using Perpetuum.Services.EventServices.EventMessages;
+using Perpetuum.Zones;
+using Perpetuum.Zones.Effects.ZoneEffects;
+
+namespace Perpetuum.Services.EventServices.EventProcessors
+{
+    public class DayTimeEventProcessor : EventProcessor<EventMessage>
+    {
+        private IZone _zone;
+        private readonly Lazy<ZoneEffect> _dayEffect;
+        private readonly Lazy<ZoneEffect> _nightEffect;
+        public DayTimeEventProcessor(IZone zone)
+        {
+            _zone = zone;
+            _dayEffect = new Lazy<ZoneEffect>(GetDayEffect);
+            _nightEffect = new Lazy<ZoneEffect>(GetNightEffect);
+        }
+
+        private ZoneEffect GetDayEffect()
```

Comment:
Maybe `CreateDayEffect` instead of `Get`? Get makes it seem like it will automatically be set or has already been set?

Reply:
Strictly speaking this *is* a `Get` operation. Creation of the `ZoneEffect` doesn't do anything by itself.